### PR TITLE
modify context length for GPTQ + version bump

### PR DESCRIPTION
**Pull request by SunMarc**
# What does this PR do ? 
This PR adds the possibility to change the max input length when using exllama backend + act_order. We also bump the required version of gptq to `0.4.2`. 
The gptq tests passed and I skipped a test because we need to wait for a release on optimum side.

### CODE REVIEW

**Review by younesbelkada — APPROVED** [review]
Amazing work @SunMarc , thanks a lot! 🔥

File: src/transformers/modeling_utils.py [hunk 1]
```diff
@@ -2570,7 +2570,11 @@ def from_pretrained(
                 raise RuntimeError("GPU is required to quantize or run quantize model.")
             elif not (is_optimum_available() and is_auto_gptq_available()):
                 raise ImportError(
-                    "Loading GPTQ quantized model requires optimum library : `pip install optimum` and auto-gptq library 'pip install auto-gptq'"
+                    "Loading GPTQ quantized model requires optimum : `pip install optimum` and auto-gptq library 'pip install auto-gptq'"
```
Comment (ArthurZucker) [inline src/transformers/modeling_utils.py:2573]:
```suggestion
                    "Loading a GPTQ quantized model requires optimum (`pip install optimum`) and auto-gptq library (`pip install auto-gptq`)"
```

**Review by ArthurZucker — COMMENTED** [review]
Thanks, left a few nits on the doc. Could you link / detail what is `exllama` and `act_order`?

File: tests/quantization/gptq/test_gptq.py [hunk 2]
```diff
@@ -236,6 +238,67 @@ class GPTQTestDeviceMapExllama(GPTQTest):
...
 
+@slow
+@require_optimum
+@require_auto_gptq
+@require_torch_gpu
+@require_accelerate
+class GPTQTestActOrderExllama(unittest.TestCase):
+    EXPECTED_OUTPUTS = set()
+    EXPECTED_OUTPUTS.add("Hello my name is Katie and I am a 20 year")
+    model_name = "TheBloke/Llama-2-7B-GPTQ"
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply tests/quantization/gptq/test_gptq.py:249]: Yeah ofc ;) . How can I be added to the org ?

File: tests/quantization/gptq/test_gptq.py [hunk 3]
```diff
@@ -236,6 +238,67 @@ class GPTQTestDeviceMapExllama(GPTQTest):
...
+        self.assertIn(self.tokenizer.decode(output_sequences[0], skip_special_tokens=True), self.EXPECTED_OUTPUTS)
+
+    def test_generate_quality(self):
+        self.check_inference_correctness(self.quantized_model)
+
+    # this test will fail until the next release of optimum
+    @pytest.mark.skip
+    def test_exllama_max_input_length(self):
+        """
+        Test if the max_input_length works with exllama + act_order
```
[the remark this answers is not in the document]
Comment (SunMarc) [reply tests/quantization/gptq/test_gptq.py]: done !

**Comment by SunMarc** [comment]
It works with the new model path :)

**Review by ArthurZucker — APPROVED** [review]
Thanks! (make sure to rebase on main before merging for the failing tests)

**Pull request merged**